pandas/core/internals/__init__.py

```diff
@@ -2725,6 +2725,11 @@ def _try_coerce_args(self, values, other):
                                 "naive Block")
             other_mask = isna(other)
             other = other.asm8.view('i8')
+        elif isinstance(other, (Timedelta, timedelta, np.timedelta64)):
+            # should be the same as TimeDeltaBlock._box_func
+            other = Timedelta(other, unit='ns')
```

Comment:
you don't need the unit

Reply:
Also don't need to check for Timedelta, since it subclasses timedelta.